Assets/Scripts/Game/UserInterface/ItemListScroller.cs

```diff
@@ -508,18 +509,21 @@ void LoadTextures(bool enhanced)
                 Texture2D redArrowsTexture = ImageReader.GetTexture(redArrowsTextureName);
                 redUpArrow = ImageReader.GetSubTexture(redArrowsTexture, upArrowRect, arrowsFullSize);
                 redDownArrow = ImageReader.GetSubTexture(redArrowsTexture, downArrowRect, arrowsFullSize);
+                GameObject.Destroy(redArrowsTexture);
```

Comment:
I'd like to understand why this destroy call is needed, or what benefits it brings. I suspect this may be an aspect of Unity game objects I am unaware of, but when I look at this code above it's getting a texture from image reader and using a locally scoped variable to hold a reference to it. Once the code block is exited this variable is out of scope and assuming there are no other references to this texture then it should simply by cleaned by the GC.

Is this destroy to expedite that? Or does it remove hidden unity references that are still held that will be preventing GC? Pretty sure the class references of subtextures wont hold a reference since the subdata is copied.

Reply:
What I realized about Unity assets (and what prompted me to start this PR) is that they don't respect the usual GC. ImageReader.GetTexture calls a Texture2D constructor which allocates a Unity Texture asset. If I don't explicitly Destroy that asset at some point then it will simply linger in memory. I've verified this by watching the texture number increase over time when opening a window that isn't a singleton. This is mitigated by a periodic call to Resources.UnloadUnusedAssets() in DaggerfallGC.cs. I'm hoping by proactively cleaning assets we can call that less often since 1. it's an expensive function and 2. the leaked memory can really pile up in between calls depending on what the player does.

Reply:
Oh wow, that's huge.. either Unity don't make that clear enough or I just somehow never realised that. Thanks for the explanation